pkgs/by-name/au/auto-patchelf/package.nix

```
@@ -0,0 +1,41 @@
+{
+  lib,
+  python3,
+}:
+
+python3.pkgs.buildPythonApplication {
+  pname = "auto-patchelf";
+  version = "0-unstable-2024-08-14";
+  format = "other";
+
+  src = lib.fileset.toSource {
+    root = ./.;
+    fileset = lib.fileset.unions [
+      ./auto-patchelf.py
+    ];
+  };
+
+  dependencies = with python3.pkgs; [
+    pyelftools
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm755 auto-patchelf.py $out/bin/auto-patchelf
+
+    runHook postInstall
+  '';
+
+  makeWrapperArgs = [
+    "--set DEFAULT_BINTOOLS $NIX_BINTOOLS"
+  ];
+
+  meta = with lib; {
+    description = "Automatically patch ELF binaries using patchelf";
+    mainProgram = "auto-patchelf";
+    license = licenses.mit;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ Scrumplex ];
+  };
```

```suggestion
  meta = {
    description = "Automatically patch ELF binaries using patchelf";
    mainProgram = "auto-patchelf";
    license = lib.licenses.mit;
    platforms = lib.platforms.any;
    maintainers = with lib.maintainers; [ Scrumplex ];
  };
```

Removes a `with lib;` and fixes `platforms` entry.

Might also be worth adding auto-patchelf's authors as mainatainers. I'll leave that up to them.

---

Limiting it to linux is probably fine, patchelf isn't used on other platforms, but we'd have to ensure it doesn't break darwin packages which use the hook on linux

---

Just so that it can be used on the BSDs once support arrives, would you mind changing it to unix?